src/ZipkinBundle/Components/HttpClient/HttpClient.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace ZipkinBundle\Components\HttpClient;
+
+use Throwable;
+use Zipkin\Tracer;
+use Zipkin\Tracing;
+use Zipkin\Propagation\Map;
+use const Zipkin\Tags\ERROR;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseStreamInterface;
+use Zipkin\SpanCustomizerShield;
+use ZipkinBundle\Components\HttpClient\ContextRetrievers\InScope;
+
+final class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var HttpClientInterface
+     */
+    private $delegate;
+
+    /**
+     * @var Tracer
+     */
+    private $tracer;
+
+    /**
+     * @var callable
+     */
+    private $injector;
+
+    /**
+     * @var HttpParser
+     */
+    private $httpParser;
+
+    /**
+     * @var ContextRetriever
+     */
+    private $contextRetriever;
+
+    public function __construct(
+        HttpClientInterface $client,
+        Tracing $tracing,
+        ContextRetriever $contextRetriever = null,
+        HttpParser $httpParser = null
+    ) {
+        $this->delegate = $client;
+        $this->tracer = $tracing->getTracer();
+        $this->injector = $tracing->getPropagation()->getInjector(new Map());
+        $this->httpParser = $httpParser ?? new DefaultHttpParser();
+        $this->contextRetriever = $contextRetriever ?: new InScope($this->tracer);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function request(string $method, string $url, array $options = []): ResponseInterface
+    {
+        if ($this->contextRetriever === null) {
+            $span = $this->tracer->nextSpan();
+        } else {
+            $context = ($this->contextRetriever)($options);
+            $span = $this->tracer->nextSpan($context);
+        }
+
+        $spanCustomizer = new SpanCustomizerShield($span);
```

Comment:
guard on noop.. don't need to make a shield on noop as you won't be parsing

Reply:
That is a good point, I am struggling to find the right balance. Is it that:
- we should avoid building a shield when span is noop (and hence not calling the customizations)?
- we should create a noop shield and don't execute the calls when it is noop

Reply:
you can inject before this line.. then it becomes a decision to parse around ` $response = $this->delegate->request` or not.. for example, in noop could you just `return  $response = $this->delegate->request`? skipping out early.. in that case the below logic is about the same

Reply:
Makes sense, I just fixed it in https://github.com/jcchavezs/zipkin-instrumentation-symfony/pull/65/files#diff-46b82fded07ac22b103293cbc8c3ebbeR64-R70